---
title: "Tutorial on genetic-ancestry inference using scRNA data"
author: "Jianing Yao"
format:
  gfm:
    toc: true           # Quarto will generate a Markdown TOC
    number-sections: true
execute:
  echo: true
  warning: false
  freeze: auto
--- 

#### This tutorial demonstrates genetic-ancestry inference for a specific HCA study (Bone marrow), using SNPs derived from scRNA-seq following the steps in `Stage1_preprocessing`.
Software requirements: 
- PLINK (1.9): https://www.cog-genomics.org/plink/
- ADMIXTURE (1.3.0): https://dalexander.github.io/admixture/download.html

### QC

In the tutorial, we consider the harmonized HGDP+1kGP dataset as a reference population dataset for genetic-ancestry inference and select six genetic-ancestry groups (Africa, America, Europe, Middle East, East Asia, and South Asia). We provided the reference dataset retaining only variants located in exonic and untranslated regions (UTRs) in `Tutorial/Stage2_ancestry_inference/HGDP_1kGP.exon_UTRS.tar.xz`. 
First, we convert the VCF file from Stage 1 into PLINK format, perform data quality control, restrict to common SNPs present in both the scRNA-seq and reference datasets, and performing genetic pruning. Here, we analyze the [bone marrow HCA dataset](https://explore.data.humancellatlas.org/projects/2a72a4e5-66b2-405a-bb7c-1e463e8febb0) as demonstration. 

```{bash qc, results='hide', message=FALSE, warning=FALSE}
# This block is command lines. 
# Set up
PLINK2="../../bin/plink1.9/plink"
# Note: Our recommended reference file is available at HGDP_1kGP.exon_UTRS.tar.xz. It contains 3,481 HGDP+1kGP individuals, and 302,006 common SNPs in coding exons and untranslated regions.
PCA_FILE="HGDP_1kGP.exon_UTRS"
tar -xJvf ../Stage2_ancestry_inference/$PCA_FILE.tar.xz
STUDY='StemCellsInChronicMyeloidLeukemia'
VCF='StemCellsInChronicMyeloidLeukemia.SNP.Filtered.SV.vcf.gz'
DIR=$STUDY
if [ ! -d "$DIR" ]; then
    mkdir -p "$DIR"
fi

#### Step 1: Convert your vcf into plink format and QC the data
$PLINK2 --const-fid 0 --vcf $VCF --geno 0.10 --make-bed --allow-extra-chr --out $DIR/$STUDY.plink.step1
$PLINK2 --bfile $DIR/$STUDY.plink.step1 --mind 0.10 --make-bed --allow-extra-chr --out $DIR/$STUDY.plink.step2
$PLINK2 --const-fid 0 --vcf $VCF --keep $DIR/$STUDY.plink.step2.fam --geno 0.10 --make-bed --allow-extra-chr --out $DIR/$STUDY.plink
rm $DIR/$STUDY.plink.step*

#### Step 2: Find rs ID and create a list with rs id in both scRNA dataset and the reference dataset
perl ../Stage2_ancestry_inference/annotate_rs_vcf.pl $DIR/$STUDY.plink $PCA_FILE
grep -v toremove $DIR/$STUDY.plink.bim | cut -f2 > $DIR/$STUDY.plink.list


#### Step 3: Merge the 2 files and perform pruning
$PLINK2 --bfile $DIR/$STUDY.plink --extract $DIR/$STUDY.plink.list --make-bed --out $DIR/$STUDY.tmp1 --allow-extra-chr
$PLINK2 --bfile $PCA_FILE         --extract $DIR/$STUDY.plink.list --make-bed --out $DIR/$STUDY.tmp2
$PLINK2 --bfile $DIR/$STUDY.tmp1  --bmerge $DIR/$STUDY.tmp2 --make-bed --allow-no-sex --out $DIR/$STUDY.HGDP_1kGP
$PLINK2 --bfile $DIR/$STUDY.HGDP_1kGP   --indep-pairwise 50 10 0.1 --out $DIR/$STUDY.HGDP_1kGP.pca
$PLINK2 --bfile $DIR/$STUDY.HGDP_1kGP   --extract $DIR/$STUDY.HGDP_1kGP.pca.prune.in --make-bed --out $DIR/$STUDY.HGDP_1kGP.pca
rm $DIR/$STUDY.tmp*
```

### PCA-Distance analysis

Next, we rely on a principal component analysis (PCA) performed on HGDP+1kGP with the projection of donors on the five first principal components (PCs). We assign each donor to the genetic-ancestry group with the smallest Euclidean distance to its centroid (method labeled PCA-Distance).

```{bash plink pca}
# This block is command lines. 
# Set up
PLINK2="../../bin/plink1.9/plink"
PCA_FILE="HGDP_1kGP.exon_UTRS"
STUDY='StemCellsInChronicMyeloidLeukemia'
DIR=$STUDY
awk '{print $1, $2, "MYID"}' $DIR/$STUDY.plink.fam > $DIR/$STUDY.HGDP_1kGP.cluster
awk '{print $1, $2, "HGDP_1kGP"}' $PCA_FILE.fam >> $DIR/$STUDY.HGDP_1kGP.cluster
$PLINK2 --bfile $DIR/$STUDY.HGDP_1kGP.pca --pca --within $DIR/$STUDY.HGDP_1kGP.cluster --pca-cluster-names HGDP_1kGP --out $DIR/$STUDY.HGDP_1kGP.pca
```

```{r pca, fig.width=8, fig.height=4, dev="png"}
library(RColorBrewer)
library(ggplot2)
library(patchwork)
library(readr)

STUDY = "StemCellsInChronicMyeloidLeukemia"
data=read.table(paste0(STUDY,"/",STUDY,".HGDP_1kGP.pca.eigenvec"),h=F)
rownames(data) = data$V2
info=read.table("info.txt",h=T,sep="\t")
rownames(info) = info$IID
mydata  = data[as.character(read.table(paste0(STUDY,"/",STUDY,".plink.fam"))[,2]),]
data_HGDP_1kGP = data[as.character(info$IID),]

topPCtokeep = 5  #let's keep the top 5 PCs
tokeep = topPCtokeep+2

#compute center of each reference population
center_afr = apply(subset(data_HGDP_1kGP[,3:tokeep],info$myREG=="Africa"),2,mean)
center_amr = apply(subset(data_HGDP_1kGP[,3:tokeep],info$myREG=="American"),2,mean)
center_eas = apply(subset(data_HGDP_1kGP[,3:tokeep],info$myREG=="East Asia"),2,mean)
center_eur = apply(subset(data_HGDP_1kGP[,3:tokeep],info$myREG=="European" | info$myREG=="Finnish"),2,mean)
center_mea = apply(subset(data_HGDP_1kGP[,3:tokeep],info$myREG=="Middle-East"),2,mean)
center_sas = apply(subset(data_HGDP_1kGP[,3:tokeep],info$myREG=="South Asia"),2,mean)
out=NULL
for (i in 1:nrow(mydata)){
    thisiid = c(
    sum((mydata[i,3:tokeep]-center_afr)**2),
    sum((mydata[i,3:tokeep]-center_amr)**2),
    sum((mydata[i,3:tokeep]-center_eas)**2),
    sum((mydata[i,3:tokeep]-center_eur)**2),
    sum((mydata[i,3:tokeep]-center_mea)**2),
    sum((mydata[i,3:tokeep]-center_sas)**2))
    thisiid = as.integer(thisiid == min(thisiid))
    out=rbind(out,thisiid)
}
colnames(out) = c("afr","amr","eas","eur","mea","sas")
rownames(out)= mydata[,2]
write.table(out,file=paste0(STUDY,"/",STUDY,".pca_center.txt"),col.names=T,row.names=F,quote=F,sep="\t")

mylegend = c("Africa", "America", "East Asia", "Europe", "Middle East", "South Asia")
mycol = c("#1B9E77","#D95F02","#7570B3","#E6AB02","#66A61E","#E7298A")
populations <- c("Africa", "American", "East Asia", "European", "Middle-East", "South Asia")
info$Color <- mycol[match(info$myREG, populations)]
#
x1=c(data_HGDP_1kGP$V3,mydata$V3); y1=c(data_HGDP_1kGP$V4,mydata$V4)
x2=c(data_HGDP_1kGP$V5,mydata$V5); y2=c(data_HGDP_1kGP$V6,mydata$V6)

df_ref <- data.frame(
  PC1 = data_HGDP_1kGP$V3, PC2 = data_HGDP_1kGP$V4,
  PC3 = data_HGDP_1kGP$V5, PC4 = data_HGDP_1kGP$V6,
  REG = factor(info$myREG, levels = populations)
)

df_my <- data.frame(
  PC1 = mydata$V3, PC2 = mydata$V4,
  PC3 = mydata$V5, PC4 = mydata$V6
)

pop_cols <- setNames(mycol, populations)

p12 <- ggplot() +
  geom_point(data = df_ref, aes(PC1, PC2, color = REG), size = 1.6, alpha = 0.7, na.rm = TRUE) +
  geom_point(data = df_my,  aes(PC1, PC2), color = "black", size = 1.8, na.rm = TRUE) +
  scale_color_manual(values = pop_cols, breaks = populations, labels = mylegend, drop = FALSE) +
  coord_cartesian(xlim = range(x1, na.rm = TRUE), ylim = range(y1, na.rm = TRUE), expand = TRUE) +
  labs(x = "PC1", y = "PC2", color = "Population") +
  theme_minimal(base_size = 11) +
  theme(legend.position = "top", panel.grid.minor = element_blank())

p34 <- ggplot() +
  geom_point(data = df_ref, aes(PC3, PC4, color = REG), size = 1.6, alpha = 0.7, na.rm = TRUE) +
  geom_point(data = df_my,  aes(PC3, PC4), color = "black", size = 1.8, na.rm = TRUE) +
  scale_color_manual(values = pop_cols, breaks = populations, labels = mylegend, drop = FALSE) +
  coord_cartesian(xlim = range(x2, na.rm = TRUE), ylim = range(y2, na.rm = TRUE), expand = TRUE) +
  labs(x = "PC3", y = "PC4", color = "Population") +
  theme_minimal(base_size = 11) +
  theme(legend.position = "top", panel.grid.minor = element_blank())

legend_row <- guides(color = guide_legend(nrow = 1, byrow = TRUE))

p12 <- p12 + legend_row + theme(legend.position = "top")
p34 <- p34 + legend_row + theme(legend.position = "top")

(p12 + p34 + plot_layout(guides = "collect", widths = c(1, 1))) &
  theme(legend.position = "top",
        legend.direction = "horizontal",
        legend.box.just = "left")
```

### ADMIXTURE analysis

We also apply ADMIXTURE in supervised mode using the HGDP+1kGP genetic-ancestry groups as reference (K=6) to estimate the proportions of ancestral populations for each donor.

```{r ADM pop}
STUDY = "StemCellsInChronicMyeloidLeukemia"

bed=read.table(paste0(STUDY,"/",STUDY,".HGDP_1kGP.fam"),h=F)[,1:2]
info=read.table("info.txt",h=T,sep="\t")
pop=rep(".",nrow(bed)); names(pop)=bed$V2

for (mypop in c("eas","amr","sas","afr","mid","eur")){
    pop[as.character(info$IID[which(info$REG==mypop)])] = mypop
}

write.table(pop,file=paste0(STUDY,"/",STUDY,".HGDP_1kGP.pca.pop"),sep="\t",quote=F,col.names=F,row.names=F)
```

```{bash run ADM}
# This block is command lines. 
STUDY='StemCellsInChronicMyeloidLeukemia'
DIR=$STUDY
cd $DIR
../../../bin/admixture_linux-1.3.0/admixture --supervised $STUDY.HGDP_1kGP.pca.bed 6
```

```{r ADM, fig.width=10, fig.height=4, dev="png"}
library(RColorBrewer)

STUDY = 'StemCellsInChronicMyeloidLeukemia'
info=read.table("info.txt",h=T,sep="\t")

admixture = read.table(paste0(STUDY,"/",STUDY,".HGDP_1kGP.pca.6.Q"),h=F)
admixture.pop = read.table(paste0(STUDY,"/",STUDY,".HGDP_1kGP.pca.pop"))
admixture.ind = read.table(paste0(STUDY,"/",STUDY,".HGDP_1kGP.pca.fam"))[,2]

colnames(admixture)=c("eur","eas","amr","sas","afr","mid")
rownames(admixture)=admixture.ind

myadmixture  = admixture[as.character(read.table(paste0(STUDY,"/",STUDY,".plink.fam"))[,2]),] # subset only mydata
toplot=myadmixture[,order(apply(myadmixture,2,mean),decreasing=T)]
mycol=c("#E6AB02","#7570B3","#D95F02","#E7298A","#1B9E77","#66A61E")
mycol=mycol[order(apply(myadmixture,2,mean),decreasing=T)]

for (i in 6:1){
    toplot=toplot[order(toplot[,i],decreasing=T),]
}

barplot(
  t(as.matrix(toplot)),
  col = mycol,
  xlab = "",
  ylab = "Admixture proportion",
  border = NA,
  xaxt = "n",
  ylim = c(0, 1)
)
```